concent_api/middleman/queue_operations.py

```diff
@@ -0,0 +1,186 @@
+from asyncio import IncompleteReadError
+from asyncio import Queue
+from asyncio import StreamReader
+from asyncio import StreamWriter
+from collections import OrderedDict  # noqa, pylint: disable=unused-import
+from logging import getLogger
+from logging import Logger
+from typing import Dict
+
+from django.conf import settings
+from django.utils.timezone import now
+
+from middleman.constants import CONNECTION_COUNTER_LIMIT
+from middleman.constants import MessageTrackerItem
+from middleman.constants import RequestQueueItem
+from middleman.constants import ResponseQueueItem
+from middleman_protocol.constants import PayloadType
+from middleman_protocol.exceptions import BrokenEscapingInFrameMiddlemanProtocolError
+from middleman_protocol.exceptions import FrameInvalidMiddlemanProtocolError
+from middleman_protocol.exceptions import PayloadTypeInvalidMiddlemanProtocolError
+from middleman_protocol.exceptions import SignatureInvalidMiddlemanProtocolError
+from middleman_protocol.message import ErrorFrame
+from middleman_protocol.message import GolemMessageFrame  # noqa, pylint: disable=unused-import
+from middleman_protocol.registry import create_middleman_protocol_message
+from middleman_protocol.stream_async import handle_frame_receive_async
+from middleman_protocol.stream_async import map_exception_to_error_code
+from middleman_protocol.stream_async import send_over_stream_async
+
+logger = getLogger()
+
+
+async def request_producer(
+    request_queue: Queue,
+    response_queue: Queue,
+    reader: StreamReader,
+    connection_id: int,
+):
+    while True:
+        try:
+            frame = await handle_frame_receive_async(reader, settings.CONCENT_PUBLIC_KEY)
+        except IncompleteReadError:
+            logger.info(f"Client has closed the connection: {connection_id}. Ending 'request_producer' coroutine")
```

Comment:
:+1: please add to pull request description that it resolves https://github.com/golemfactory/concent/issues/679

Reply:
done